### PR TITLE
feat: :sparkles: Converted javascript to typescript

### PR DESCRIPTION
**Pull request by KiveshVirasamy**
Changed the following files to typescript: main.js, dom-manipulation.js, leaflet-services.js. I also removed used files

### CODE REVIEW

File: src/dom-manipulation.ts [hunk 1]
```diff
@@ -1,9 +1,25 @@
 // Import Leaflet and utility functions
-import * as LeafletServices from "./services/leaflet-services.js";
-import * as Utils from "./utils/utilities.js";
+import * as LeafletServices from "./services/leaflet-services";
+import * as Utils from "./utils/utilities";
+
+// Define the Flight type
+type Flight = [
```
Comment (CharelMarais) [inline src/dom-manipulation.ts:6]: I would say you need to move your type of declaration to a separate

File: src/dom-manipulation.ts [hunk 2]
```diff
@@ -12,7 +28,7 @@ export function addFlightInfoToContainer(flight) {
   // Convert the speed from m/s to km/h using utility function
   const speed = Utils.convertMPSToKPH(flight[11] ?? 0.0);
```
Comment (CharelMarais) [inline src/dom-manipulation.ts:29]: Just make sure your variables have set types and not any.

File: src/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
```
Comment (CharelMarais) [inline src/tsconfig.json:2]: You should add "noImplicitAny": true to check for any Implicite Any's

**Review by CharelMarais — CHANGES_REQUESTED** [review]
Look good just double check there are no Implicite any's you can view my comments.

File: src/dom-manipulation.ts (hunk 1 above)
Comment (KiveshVirasamy) [reply src/dom-manipulation.ts:6]: Okay will do for the separation of concerns?

File: src/dom-manipulation.ts (hunk 2 above)
Comment (KiveshVirasamy) [reply src/dom-manipulation.ts:29]: I totally overlooked that one on it now.

File: src/tsconfig.json (hunk 3 above)
Comment (KiveshVirasamy) [reply src/tsconfig.json:2]: Okay adding now. Thanks man.

**Comment by theadley** [comment]
Most of these are unresolved, yet merged anyway. There are still type errors and you have loads of explicit and implicit `any` types all over the show. Why merge?

**Pull request merged**